src/repo/zcl_abapgit_file_status.clas.locals_imp.abap

```diff
@@ -0,0 +1,301 @@
+
+    " TODO optimize ?
+    " sort by obj, path
+    " loop, and compare to first object record
+
+    " Collect object index
+    lt_res_sort = it_results.
+    SORT lt_res_sort BY obj_type ASCENDING obj_name ASCENDING.
+
+    LOOP AT it_results ASSIGNING <ls_result> WHERE NOT obj_type IS INITIAL AND packmove = abap_false.
+
+      IF NOT ( <ls_result>-obj_type = ls_item-obj_type
+          AND <ls_result>-obj_name = ls_item-obj_name ).
+        APPEND INITIAL LINE TO lt_item_idx ASSIGNING <ls_result_idx>.
+        <ls_result_idx>-obj_type = <ls_result>-obj_type.
+        <ls_result_idx>-obj_name = <ls_result>-obj_name.
+        <ls_result_idx>-path     = <ls_result>-path.
+        MOVE-CORRESPONDING <ls_result> TO ls_item.
+      ENDIF.
+
+    ENDLOOP.
+
+    LOOP AT it_results ASSIGNING <ls_result>
+      WHERE NOT obj_type IS INITIAL AND obj_type <> 'DEVC' AND packmove = abap_false.
+
+      READ TABLE lt_item_idx ASSIGNING <ls_result_idx>
+        WITH KEY obj_type = <ls_result>-obj_type obj_name = <ls_result>-obj_name
+        BINARY SEARCH. " Sorted above
+
+      IF sy-subrc <> 0 OR <ls_result>-path <> <ls_result_idx>-path. " All paths are same
+        mi_log->add( iv_msg = |Files for object { <ls_result>-obj_type } {
+                              <ls_result>-obj_name } are not placed in the same folder|
+                     iv_type = 'W' ).
+      ENDIF.
+
+    ENDLOOP.
+
+  ENDMETHOD.
+
+  METHOD check_multiple_files.
+
+    DATA:
+      lt_res_sort LIKE it_results,
+      ls_file     TYPE zif_abapgit_git_definitions=>ty_file_signature.
+
+    FIELD-SYMBOLS <ls_result> LIKE LINE OF it_results.
+
+    lt_res_sort = it_results.
+    SORT lt_res_sort BY filename ASCENDING.
+
+    LOOP AT lt_res_sort ASSIGNING <ls_result> WHERE obj_type <> 'DEVC' AND packmove = abap_false.
+      IF <ls_result>-filename IS NOT INITIAL AND <ls_result>-filename = ls_file-filename.
+        mi_log->add( iv_msg  = |Multiple files with same filename, { <ls_result>-filename }|
+                     iv_type = 'W' ).
+      ENDIF.
+
+      IF <ls_result>-filename IS INITIAL.
+        mi_log->add( iv_msg  = |Filename is empty for object { <ls_result>-obj_type } { <ls_result>-obj_name }|
+                     iv_type = 'W' ).
+      ENDIF.
+
+      MOVE-CORRESPONDING <ls_result> TO ls_file.
+    ENDLOOP.
+
+  ENDMETHOD.
+
+
+  METHOD check_namespace.
+
+    DATA:
+      li_namespace TYPE REF TO zif_abapgit_sap_namespace,
+      lv_namespace TYPE namespace,
+      lt_namespace TYPE TABLE OF namespace.
+
+    FIELD-SYMBOLS <ls_result> LIKE LINE OF it_results.
+
+    " Collect all namespaces based on name of xml-files
+    LOOP AT it_results ASSIGNING <ls_result>.
+      FIND REGEX '^#([a-zA-Z0-9]+)#.*\..*\.xml$' IN <ls_result>-filename SUBMATCHES lv_namespace.
+      IF sy-subrc = 0.
+        lv_namespace = '/' && to_upper( lv_namespace ) && '/'.
+        COLLECT lv_namespace INTO lt_namespace.
+      ENDIF.
+    ENDLOOP.
```

Comment:
needs to get fixed since it does not work with AFF `json` which also uses `( )` instead of `#` for namespace in filename (was wrong before but not noticed)

Reply:
see `zcl_abapgit_filename_logic->file_to_object`

Reply:
Probably yes, but I'm far from the topic. I suggest keeping it in this PR as it. In "part3" I'll do some adjustments to checks and I'll create the branch in this repo so you could commit the change as well.

Reply:
keep as is. reminder: https://github.com/abapGit/abapGit/issues/6126